### ISSUE TITLE
region commit state jump backward 

### ISSUE THREAD

**Issue by lilinghai** [pingcap/tiflash#2576]
```
[2021/08/04 22:56:26.752 +08:00] [FATAL] [lib.rs:462] ["[region 2016216] 2016217 commit state jump backward (277150, 13) -> (276191, 13)"] [backtrace="stack backtrace:
   0: tikv_util::set_panic_hook::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at library/std/src/panicking.rs:595
   2: std::panicking::begin_panic_handler::{{closure}}
             at library/std/src/panicking.rs:497
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at library/std/src/sys_common/backtrace.rs:141
   4: rust_begin_unwind
             at library/std/src/panicking.rs:493
   5: std::panicking::begin_panic_fmt
             at library/std/src/panicking.rs:435
   6: raftstore::store::fsm::apply::ApplyFsm<EK>::handle_tasks
   7: <raftstore::store::fsm::apply::ApplyPoller<EK,W> as batch_system::batch::PollHandler<raftstore::store::fsm::apply::ApplyFsm<EK>,raftstore::store::fsm::apply::ControlFsm>>::handle_normal
   8: std::sys_common::backtrace::__rust_begin_short_backtrace
   9: core::ops::function::FnOnce::call_once{{vtable.shim}}
  10: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/16bf626a31cb5b121d0bca2baa969b4f67eb0dab/library/alloc/src/boxed.rs:1546
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/16bf626a31cb5b121d0bca2baa969b4f67eb0dab/library/alloc/src/boxed.rs:1546
      std::sys::unix::thread::Thread::new::thread_start
             at library/std/src/sys/unix/thread.rs:71
  11: start_thread
  12: __clone
"] [location=/build/tics/contrib/tiflash-proxy/components/raftstore/src/store/fsm/apply.rs:3326] [thread_name=apply-3]
```

**Comment by flowbehappy** [comment]
Related issue: https://github.com/tikv/tikv/issues/11396

**Comment by CalvinNeo** [comment]
Fixed in 5.1 5.2 5.3 5.4 6.1
